src/core/string.c

```diff
@@ -124,33 +146,10 @@ int string_from_int(uint8_t *dst, int value, int force_plus_sign)
 
 int string_compare_case_insensitive(const char *a, const char *b)
 {
-    while (*a && *b) {
-        int aa = tolower(*a);
-        int bb = tolower(*b);
-        if (aa != bb) {
-            return aa - bb;
-        }
-        ++a;
-        ++b;
-    }
-    if (*a) {
-        return 1;
-    }
-    if (*b) {
-        return -1;
-    }
-    return 0;
+    return strcasecmp(a, b);
 }
 
-int string_equals(const uint8_t *a, const uint8_t *b)
+int string_compare_case_insensitive_prefix(const char *full_string, const char *prefix, int prefix_len)
 {
-    while (*a && *b && *a == *b) {
-        ++a;
-        ++b;
-    }
-    if (*a == 0 && *b == 0) {
-        return 1;
-    } else {
-        return 0;
-    }
+    return strncasecmp(full_string, prefix, prefix_len);
```

Comment:
I know you did this to avoid code duplication, but I really don't like the use of the non-standard `strcasecmp` and `strncasecmp`...

Reply:
I did this because the existing implementation of `string_compare_case_insensitive` doesn't work for non-ascii (multibyte) characters, always placing them above the files starting with ascii characters.

Now that I think of it: it's also weird that these two functions are the only ones in this file that operate on `char *`, while the others are for `uint8_t *`. Since all usages of `string_compare_case_insensitive*` are related to file names, I think it makes sense to move these functions to `file_manager` and handle MSVC/Windows in the proper way there. What do you think?

Reply:
Lovely idea! But change it to `compare_file_name` out something...

Reply:
I just moved the functions to `file_manager`, looks much cleaner!

Could you test this on windows with both the mingw and msvc builds?
On my Mac it lists a file starting with "ä" under "a" now, which is pretty nice, but that may just be a platform thing.

Reply:
Both mingw and msvc compile and run.

However, unlike mac, `a` won't match `á`. For some reason, `à` is also not matching `à`, although `é` is matching `é`.

Reply:
Another comment: When deleting chars (using backspace) the file manager keeps updating the filter and thus moving. Shouldn't the filter only apply when the user inputs chars, but not when removing them?

Reply:
Good suggestion. I wasn't sure what the user would expect when deleting characters.